src/OneExplorer.ts

```diff
@@ -226,30 +227,68 @@ export class OneExplorer {
 // menu handler
 //
 
-export {handleRunOnecc};
+import {EventEmitter} from 'events';
 
-/**
- * Function called when onevscode.run-cfg is called (when user clicks 'Run' on cfg file).
- * @param cfgUri uri of cfg file
- */
-function handleRunOnecc(cfgUri: vscode.Uri, logger: Logger) {
-  const toolRunner = new ToolRunner(logger);
-
-  // TODO Refine later
-  const resolve = function(value: string) {
-    console.log('Running onecc was successful!');
-  };
-  const reject = function(value: string) {
-    console.log('Running onecc failed!');
-  };
-
-  const toolArgs = new ToolArgs('-C', cfgUri.fsPath);
-  const cwd = path.dirname(cfgUri.fsPath);
-  let oneccPath = toolRunner.getOneccPath();
-  if (oneccPath === undefined) {
-    throw new Error('Cannot find installed onecc');
-  }
-
-  const runner = toolRunner.getRunner('onecc', oneccPath, toolArgs, cwd);
-  runner.then(resolve).catch(reject);
+class OneccRunner extends EventEmitter {
+  private startRunningOnecc: string = 'START_RUNNING_ONECC';
+  private finishedRunningOnecc: string = 'FINISHED_RUNNING_ONECC';
+
+  constructor(private cfgUri: vscode.Uri, private logger: Logger) {
+    super();
+  }
+
+  /**
+   * Function called when onevscode.run-cfg is called (when user clicks 'Run' on cfg file).
+   */
+  public run() {
+    const toolRunner = new ToolRunner(this.logger);
+
+    this.on(this.startRunningOnecc, this.onStartRunningOnecc);
+    this.on(this.finishedRunningOnecc, this.onFinishedRunningOnecc);
+
+    const toolArgs = new ToolArgs('-C', this.cfgUri.fsPath);
+    const cwd = path.dirname(this.cfgUri.fsPath);
+    let oneccPath = toolRunner.getOneccPath();
+    if (oneccPath === undefined) {
+      throw new Error('Cannot find installed onecc');
+    }
+
+    const runnerPromise = toolRunner.getRunner('onecc', oneccPath, toolArgs, cwd);
+    this.emit(this.startRunningOnecc, runnerPromise);
+  }
+
+  private onStartRunningOnecc(runnerPromise: Promise<string>) {
+    const progressOption: vscode.ProgressOptions = {
+      location: vscode.ProgressLocation.Notification,
+      title: `Running onecc --config ${this.cfgUri.fsPath}`,
```

Comment:
```suggestion
      title: `Running: 'onecc --config ${this.cfgUri.fsPath}'`,
```
Mayby this will enhance readability